### ISSUE TITLE
Definition of BAD EPOCHS

### ISSUE THREAD

**Issue by jayurbain** [flairNLP/flair#1429]
After reading this issue: https://github.com/flairNLP/flair/issues/609 ,  I'm still a little confused about the meaning of BAD EPOCHS. I thought BAD EPOCHS for any EPOCH meant the cumulative number of BAD EPOCHS up through and including the current EPOCH.

But I find in the training output example below, that the value of BAD EPOCHs vacillates. i.e., 3, 4, 1, 2,.... Any guidance would be appreciated.

-------------------------------
2020-02-16 09:39:56,016 epoch 28 - iter 1/1 - loss 15.28764820 - samples/sec: 30.19
2020-02-16 09:39:56,076 ----------------------------------------------------------------------------------------------------
2020-02-16 09:39:56,077 EPOCH 28 done: loss 15.2876 - lr 0.0016
2020-02-16 09:39:56,143 DEV : loss 10.322088241577148 - score 0.0
2020-02-16 09:39:56,146 BAD EPOCHS (no improvement): 3
2020-02-16 09:40:00,204 ----------------------------------------------------------------------------------------------------
2020-02-16 09:40:01,075 epoch 29 - iter 1/1 - loss 15.28552628 - samples/sec: 36.85
2020-02-16 09:40:01,128 ----------------------------------------------------------------------------------------------------
2020-02-16 09:40:01,129 EPOCH 29 done: loss 15.2855 - lr 0.0016
2020-02-16 09:40:01,188 DEV : loss 10.319738388061523 - score 0.0
Epoch    29: reducing learning rate of group 0 to 7.8125e-04.
2020-02-16 09:40:01,190 BAD EPOCHS (no improvement): 4
2020-02-16 09:40:05,051 ----------------------------------------------------------------------------------------------------
2020-02-16 09:40:06,019 epoch 30 - iter 1/1 - loss 15.26811409 - samples/sec: 33.13
2020-02-16 09:40:06,070 ----------------------------------------------------------------------------------------------------
2020-02-16 09:40:06,071 EPOCH 30 done: loss 15.2681 - lr 0.0008
2020-02-16 09:40:06,132 DEV : loss 10.313851356506348 - score 0.0
2020-02-16 09:40:06,133 BAD EPOCHS (no improvement): 1
2020-02-16 09:40:10,052 ----------------------------------------------------------------------------------------------------
2020-02-16 09:40:11,026 epoch 31 - iter 1/1 - loss 15.11200619 - samples/sec: 32.93
2020-02-16 09:40:11,071 ----------------------------------------------------------------------------------------------------
2020-02-16 09:40:11,072 EPOCH 31 done: loss 15.1120 - lr 0.0008
2020-02-16 09:40:11,138 DEV : loss 10.311601638793945 - score 0.0
2020-02-16 09:40:11,140 BAD EPOCHS (no improvement): 2
2020-02-16 09:40:15,141 ----------------------------------------------------------------------------------------------------
2020-02-16 09:40:16,233 epoch 32 - iter 1/1 - loss 15.39288616 - samples/sec: 29.35
2020-02-16 09:40:16,285 ----------------------------------------------------------------------------------------------------
2020-02-16 09:40:16,286 EPOCH 32 done: loss 15.3929 - lr 0.0008
2020-02-16 09:40:16,358 DEV : loss 10.308276176452637 - score 0.0
2020-02-16 09:40:16,361 BAD EPOCHS (no improvement): 3
2020-02-16 09:40:20,714 ----------------------------------------------------------------------------------------------------
Thanks,
Jay

**Comment by djstrong** [comment]
If the learning rate decreases then it starts counting from 0.

**Comment by jayurbain** [comment]
Thank you.
